pages/api/roles/index.ts

```diff
@@ -66,13 +67,41 @@ async function handler(req: NextApiRequest, res: NextApiResponse) {
   }
 }
 
-// TODO: Optional Search query. Will be done in another PR to unblock others.
-
 async function handleGet(req: NextApiRequest, res: NextApiResponse<ApiResponse<RoleListData[]>>) {
+  const { companyId, searchQuery } = req.query as RoleQueryParams;
+
+  const queryCompanyId = typeof companyId == 'string' && isInteger(companyId) ? Number(companyId) : undefined;
+
+  const searchWords = typeof searchQuery == 'string' && !isEmpty(searchQuery) ? searchQuery.split(/\s+/) : [];
```

Comment:
```suggestion
  const searchWords = typeof searchQuery == 'string' && !isEmpty(searchQuery) ? searchQuery.trim().split(/\s+/) : [];
```

So that we don't get empty strings at the start or end after splitting.